matchbox_socket/src/webrtc_socket/socket.rs

```diff
@@ -179,6 +183,19 @@ impl WebRtcSocketBuilder {
         self.config.keep_alive_interval = interval;
         self
     }
+
+    /// Sets authentication on request to the signaling server.
+    pub fn basic_authentication(
+        mut self,
+        username: impl Into<String>,
+        password: impl Into<String>,
```

Comment:
I'd like to explicitly avoid performing auth with a `username`+`password`. This makes the signalling server an auth server which adds a whole heap of complications, I would much rather we had it act as an auth client and delegate the actual authentication to a purpose build server via one of the Oauth flows

Reply:
Yes, we would only perform authorisation, not authentication. The authentication could be done by an external service (e.g. steam) or by including an oath library such as [oauth2](https://docs.rs/oauth2/latest/oauth2/). This way we can avoid both having to design an secure auth flow, and having to write code that handles user credentials